plugins/dualGovernance/hooks/useCanCreateProposal.tsx

```diff
@@ -44,6 +49,25 @@ export function useCanCreateProposal() {
     ],
   });
 
+  // Check if PROPOSER_PERMISSION is granted to the current wallet
+  const {
+    data: hasCreatePermissionData,
+    refetch: hasCreatePermissionRefetch,
+    status: hasCreatePermissionStatus,
+  } = useReadContract({
+    chainId: PUB_CHAIN.id,
+    address: daoAddress,
+    abi: DaoAbi,
+    functionName: "hasPermission",
+    // where, who, permissionId, data
+    args: [
+      PUB_DUAL_GOVERNANCE_PLUGIN_ADDRESS,
+      address,
+      keccak256(toHex("PROPOSER_PERMISSION")),
+      "0x",
+    ],
+  });
```

Comment:
I'm admittedly missing some context, but why can't we do something like this so that we don't need to manually refetch this query in the useEffect?

```suggestion
  const {
    data: hasCreatePermissionData,
    refetch: hasCreatePermissionRefetch,
    status: hasCreatePermissionStatus,
  } = useReadContract({
    chainId: PUB_CHAIN.id,
    address: daoAddress,
    abi: DaoAbi,
    functionName: "hasPermission",
    // where, who, permissionId, data
    args: [
      PUB_DUAL_GOVERNANCE_PLUGIN_ADDRESS,
      address,
      keccak256(toHex("PROPOSER_PERMISSION")),
      "0x",
      query: {enabled: !!address && !!daoAddress}
    ],
  });
```